dandi/files.py

```diff
@@ -518,21 +518,25 @@ def get_validation_errors(
         else:
             # make sure that we have some basic metadata fields we require
             try:
-                meta = get_metadata(self.filepath)
+                errors.extend(
+                    [
+                        error.message
+                        for error in inspect_nwb(
+                            nwbfile_path=self.filepath,
+                            checks=_required_nwb_checks,
+                        )
+                    ]
```

Comment:
Note two items here:

(a) I notice in your validation API you expect the output to be a list of strings; our native output from the Inspector is a sophisticated [data class](https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/nwbinspector/register_checks.py#L35-L75) that also tracks a bunch of other contextual information, the most immediately important of which here is likely the `message`, which is why I'm only pulling out that item.

If desired, we could figure out some sort of 'nice' f-string that incorporates other relevant attributes. But for now, the form of the output can be seen in the modified and added tests for this PR

(b) All the highest-level `inspect` functionalities such as `inspect_nwb` used here are generators that yield data streams of messages; this iteration is being collapsed in this stage similar to the existing functionality, but if desired we could look into yielding here as well.

Reply:
We definitely should refactor validation output to support more informative structured records. For the purpose of this PR let's KISS **but** it relates to #896 for adding BIDS validation.

But I thought that inspector just provides "hints" or "warnings" not "error" class of notifications. 
In refactoring we might provide generalization over types of records known to bids-validator and nwb inspector.  Filed https://github.com/dandi/dandi-cli/issues/943 to start dialog on that topic. Haven't looked yet into nwbinspector records

Reply:
> But I thought that inspector just provides "hints" or "warnings" not "error" class of notifications.

Precisely; the Inspector only returns messages, we don't/can't distinguish errors vs. warnings most of the time. The behavior occurring here is to basically consider ANY message returned as an error to prevent validation+upload.

Reply:
Well, technically, there are two types of explicit errors that we can and do catch, (i) actual Python errors that occur during the process, these get returned as traceback messages but inspection proceeds, and (ii) pynwb validation, but we're planning to disable that here according the other discussion.

You could *technically* consider anything else from the inspector a 'warning', but the intended use here is for things like critical missing metadata to be considered as errors in the context of validation/upload.

Reply:
the question here -- should list comprehension be adjusted to only consider errors and skipping any hints etc, or those wouldn't be included ATM any ways? Or should we specify that config should be "dandi"? (I guess it relates to discussion on "tag"ing which checks to run for dandi [below](https://github.com/dandi/dandi-cli/pull/941#discussion_r831507352))

Reply:
@yarikoptic we are using the nwb inspector in two different modes.
1. provide a report, tailored to the needs of dandi, recommending changes to the file
2. validating the file to ensure all essential requirements are met.

this PR is for mode 2. Mode 1 is taken care of by this PR: https://github.com/dandi/handbook/pull/40 and requires no changes to the dandi cli